### PR TITLE
feat (project landing page): download metadata in JSON format (DSP-1184)

### PR DESCRIPTION
**Pull request by waychal**
resolves DSP-1184

It adds the functionality on project landing page to download metadata in JSON format.

### CODE REVIEW

**Review by flavens — COMMENTED** [review]
why did you push your package-lock?

**Comment by waychal** [comment]
> why did you push your package-lock?

I accidentally added it. I have reverted back the changes.

File: src/app/project/board/board.component.ts [hunk 1]
```diff
@@ -262,6 +262,22 @@ export class BoardComponent implements OnInit {
         this.keywords = this.projectMetadata['keywords'];
     }
 
+    // download metadata
+    downloadMetadata() {
```
Comment (flavens) [inline src/app/project/board/board.component.ts:266]: question: will you use this method only for JSON or as a generic one for the download format?

Comment (waychal) [reply src/app/project/board/board.component.ts:266]: Once the endpoints for other formats will be available, this method will be converted into generic one where format will be passed as a parameter.

File: src/app/project/board/board.component.ts [hunk 2]
```diff
@@ -262,6 +262,22 @@ export class BoardComponent implements OnInit {
...
+        const blob: Blob = new Blob([JSON.stringify(this.metadata)], {type: 'application/json'});
+        const fileName: string = 'metadata.json';
+        const objectUrl: string = URL.createObjectURL(blob);
+        const a: HTMLAnchorElement = document.createElement('a') as HTMLAnchorElement;
```
Comment (mpro7) [inline src/app/project/board/board.component.ts:270]: You don't need to annotate type if the value of properties it's already initialised. But I'm not sure which convention we follow in DSP-APP. It's generally not bad, but redundant.

**Review by flavens — APPROVED** [review]

File: src/app/project/board/board.component.ts (hunk 1 above)
Comment (mpro7) [reply src/app/project/board/board.component.ts:266]: I think there will be no separate endpoint for downloads, but just different parameter to pass to the GET method. But this needs to be confirmed with BE team.

File: src/app/project/board/board.component.ts (hunk 2 above)
Comment (waychal) [reply src/app/project/board/board.component.ts:270]: I also don't know. May be @flavens can comment it and I will use it next time.

File: src/app/project/board/board.component.ts (hunk 1 above)
Comment (waychal) [reply src/app/project/board/board.component.ts:266]: We should create issue for adding endpoints for different formats in backend and prioritise it.

**Pull request merged**